File: Binary-Search.py
1class Solution(object):
2    def search(self, nums, target):
3        left, right = 0, len(nums) - 1
4        
5        while left <= right:
6            mid = (left + right) // 2
7            
8            if nums[mid] == target:
9                return mid
10            elif nums[mid] < target:
11                left = mid + 1
12            else:
13                right = mid - 1
14        
15        return -1
16        

File: Binary-Search.java
1class Solution {
2
3    public int search(int[] nums, int target) {
4        int left = 0;
5        int right = nums.length - 1;
6
7        while (left <= right) {
8            int mid = (left + right) / 2;
9
10            if (nums[mid] == target) {
11                return mid;
12            } 
13            else if (nums[mid] < target) {
14                left = mid + 1;
15            } 
16            else {
17                right = mid - 1;
18            }
19        }
20
21        return -1;   // required return if target not found
22    }
23
24}